### PR TITLE
Adding Tomcat plugin.

### PR DESCRIPTION
**Pull request by mlindes**
This plugin resolves open issue  #763 (https://github.com/influxdata/telegraf/issues/763)

### Required for all PRs:
- [ ] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

### CODE REVIEW

**Comment by mlindes** [comment]
Any chance this will make the 1.3 release?

**Comment by danielnelson** [comment]
No, I released a rc for 1.3 yesterday: #2733.

**Comment by nitin02** [comment]
Hi, Is this plugin release added to the next milestone? Which version will this plugin be available?

**Comment by toni-moreno** [comment]
We would like to test it!!

File: plugins/inputs/tomcat/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+```toml
+# A Telegraf plugin to collect tomcat metrics.
+[[inputs.tomcat]]
+  # A Tomcat status URI to gather stats.
+  # Default is "http://127.0.0.1:8080/manager/status/all?XML=true".
+  url = http://127.0.0.1:8080/manager/status/all?XML=true
+  # Cedentials for status URI.
+  # Default is tomcat/s3cret.
+  username = tomcat
+  password = s3cret
```
Comment (danielnelson) [inline plugins/inputs/tomcat/README.md]: These strings need to be quoted so they are valid toml.

File: plugins/inputs/tomcat/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+### Measurements & Fields:
+
+<optional description>
```
Comment (danielnelson) [inline plugins/inputs/tomcat/README.md]: Remove this line

File: plugins/inputs/tomcat/tomcat.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+var sampleconfig = `
+  ## A Tomcat status URI to gather stats.
+  ## Default is "http://127.0.0.1:8080/manager/status/all?XML=true".
+  url = http://127.0.0.1:8080/manager/status/all?XML=true
+	## Cedentials for status URI.
+  ## Default is tomcat/s3cret.
+	username = tomcat
+	password = s3cret
```
Comment (danielnelson) [inline plugins/inputs/tomcat/tomcat.go]: Quote toml strings here too.

**Comment by mlindes** [comment]
Quotes added. CircleCI complained about a zookeeper task. Not sure that's related to this pr.. I can follow-up if it is, though.

**Comment by danielnelson** [comment]
I think if you rebase it will fix the zookeeper issue.

**Comment by mlindes** [comment]
I don't really see how a rebase would solve this CircleCI error:

```
sudo service zookeeper stop
zookeeper: unrecognized service
sudo service zookeeper stop returned exit code 1
Action failed: sudo service zookeeper stop
```

Can you re-trigger the build to try it again? Permissions don't allow me to do that.
Otherwise, which tag would you like me to rebase off of. The first (successful) build was built from the 1.2.1 tag - which was the most current at the time of the PR.

**Comment by mlindes** [comment]
Oh sorry, I spoke too soon. I didn't notice the `circle.yml` file is totally different in the current base.
Looks like the latest change to `circle.yml` was after the 1.3.5 tag.  Maybe I can just pull that one file. Thoughts?

**Comment by danielnelson** [comment]
I haven't tried that ever but when merge I will do a "Squash and merge" so as long as it does not introduce unwanted diffs it's fine.

**Comment by mlindes** [comment]
ok. I rebased off master, and tests now pass. The rebase introduced a few commits... but there hasn't been a tag or version release to use as a rebase point since the latest changes to the `circle.yml` file.

**Comment by danielnelson** [comment]
You will need to remove the unrelated commits during the rebase (use the `-i` flag).

**Comment by mlindes** [comment]
Accidentally closed this while trying to get our fork in sync.
Please check out the new PR - https://github.com/influxdata/telegraf/pull/3112

**Pull request closed without merging**